hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -225,6 +129,16 @@ public void addActionDefinition(String tenantId, ActionDefinition actionDefiniti
         if (isEmpty(actionDefinition.getProperties())) {
             throw new IllegalArgumentException("Properties must be not null");
         }
+        String plugin = actionDefinition.getActionPlugin();
+        if (!getActionPlugins().contains(plugin)) {
+            throw new IllegalArgumentException("Plugin: " + plugin + " is not deployed");
+        }
+        Set<String> pluginProperties = getActionPlugin(plugin);
+        actionDefinition.getProperties().keySet().stream().forEach(property -> {
+            if (!pluginProperties.contains(property)) {
+                throw new IllegalArgumentException("Property: " + property + " is not valid on plugin: " + plugin);
+            }
```

Comment:
Plugins declare which are supported properties.
This check is to ensure that a property defined in an action are present in the plugin.
No more validation defined (types or values).
I think an exception is ok to force to use correct properties, if not validation has not sense at all (I think).
The REST API is modified, so, when no properties defined, a "Bad request" response is sent instead of "Internal error".
Perhaps in the future we can add validation at value level, but that is more complex as it might required more changes in the API and more communication between plugins are engine.